opensuse_buildservice/debian/preinst

```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+# preinst script for notqmail
+#
+# see: dh_installdeb(1)
+
+set -e
+qmaildir=/var/qmail
+logdir=/var/log/svc
+
+# summary of how this script can be called:
+#        * <new-preinst> `install'
+#        * <new-preinst> `install' <old-version>
+#        * <new-preinst> `upgrade' <old-version>
+#        * <old-preinst> `abort-upgrade' <new-version>
+# for details, see http://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+
+case "$1" in
+    install|upgrade)
+	rm -f /tmp/notqmail.install /tmp/notqmail.upgrade
```

Comment:
I find your handling of files in /tmp a bit scary. Even if you remove notqmail.upgrade here I can still create it while the scripts are running to influence what happens later.

Even worse, I can create the logfile as a symlink to something owned by root and can then corrupt it. Does Debian hide all errors of these scripts? Is install-or-upgrade not passed in a sensible way to postinst?

If you really need to communicate these things via file I suggest putting them in /var/qmail, this is something this package controls and that is not writable by the user.

Reply:
I think in case of upgrade argc is 3 and postinst is called with two arguments where argv1 is configure and argv2 is the version of previous package
in case of installation argc is 2 with argv1 as configure
I will try this. In that case use of files as flags can be eliminated

Reply:
Even the creation of /tmp/notqmail.log should be removed. It will have the same risk.

Reply:
done